src/v/container/include/container/interval_set.h

```diff
@@ -34,7 +34,8 @@ class interval_set {
         T start;
 
         // Exclusive.
-        T end;
+        // It's important that this not be used in any comparisons.
+        mutable T end;
```

Comment:
@andrwng

Reply:
std::btree_set in newer abseil returns const iterators (like std::set - https://en.cppreference.com/w/cpp/container/set/begin). the rationale for this is that the container is indexing a T but the compiler can't determine which parts of T are used in the comparison function.

Reply:
we could also remove and then re-insert to make compiler happy

Reply:
Thanks! It'd be great to add this context to the commit message here or of the final commit

Reply:
done